platform/darwin/src/MGLMapSnapshotter.mm

```diff
@@ -115,182 +116,192 @@ - (void)startWithQueue:(dispatch_queue_t)queue completionHandler:(MGLMapSnapshot
                     format:@"Already started this snapshotter."];
     }
     
-    _loading = true;
+    self.loading = true;
     
-    dispatch_async(queue, ^{
-        _snapshotCallback = std::make_unique<mbgl::Actor<mbgl::MapSnapshotter::Callback>>(*mbgl::Scheduler::GetCurrent(), [=](std::exception_ptr mbglError, mbgl::PremultipliedImage image, mbgl::MapSnapshotter::Attributions attributions, mbgl::MapSnapshotter::PointForFn pointForFn) {
-            _loading = false;
+    __weak __typeof__(self) weakSelf = self;
+    _snapshotCallback = std::make_unique<mbgl::Actor<mbgl::MapSnapshotter::Callback>>(*mbgl::Scheduler::GetCurrent(), [=](std::exception_ptr mbglError, mbgl::PremultipliedImage image, mbgl::MapSnapshotter::Attributions attributions, mbgl::MapSnapshotter::PointForFn pointForFn) {
+        __typeof__(self) snapshotter = weakSelf;
+        snapshotter.loading = false;
+        
+        
+        if (mbglError) {
+            NSString *description = @(mbgl::util::toString(mbglError).c_str());
+            NSDictionary *userInfo = @{NSLocalizedDescriptionKey: description};
+            NSError *error = [NSError errorWithDomain:MGLErrorDomain code:MGLErrorCodeSnapshotFailed userInfo:userInfo];
             
-            NSMutableArray *infos = [NSMutableArray array];
-
+            // Dispatch result to origin queue
+            dispatch_async(queue, ^{
+                completion(nil, error);
+            });
+        } else {
 #if TARGET_OS_IPHONE
-            CGFloat fontSize = [UIFont smallSystemFontSize];
-            UIColor *attributeFontColor = [UIColor blackColor];
+            MGLImage *mglImage = [[MGLImage alloc] initWithMGLPremultipliedImage:std::move(image) scale:weakSelf.options.scale];
```

Comment:
Use `snapshotter` (or `strongSelf`) instead of `weakSelf` throughout this lambda. Otherwise, `snapshotter` only strongly captures the snapshotter object for as long as that variable is needed within this lambda.